### ISSUE TITLE
Handling server-side exceptions while in-place editing -- is it supported?

### ISSUE THREAD

**Issue by artvolk** [littleK0i/jqGridPHP#2]
Привет ещё раз, issue #2, похоже тоже будет моя :)

Пытаюсь реализовать in-place редактирование одного поля в таблице. Сделал так:

На PHP стороне для колонки:

``` PHP
'editable' => true
```

Перегрузил метод, выбрасываю в нём исключение

``` PHP
protected function opEdit($id, $upd)
{
    throw new jqGrid_Exception('Значение должно быть в диапазоне от 0 до 1!');
}
```

При редактировании значения в ячейке в Firebug вижу, что в ответ пришел JSON-объект с ошибкой:

``` javascript
{ "error": "1", "error_msg": "Значение должно быть в диапазоне от 0 до 1!", "error_code": "0", "error_data": null, "error_type": "common" }
```

Но на клиенте сообщение об ошибке не отображается. Скорее всего дело в том, что вот это событие не вызывается при in-place редактировании:

``` javascript
'loadComplete'  : function(data)
{
    $.jgrid.ext.errorHandler(data);
}
```

Вопросы:
1. Правильно ли я все понимаю, или где-то что-то пропустил в документации и все должно работать из коробки?
2. В демках есть только редактирование с помощью всплывающего диалога, которое вызывается кнопкой на панели навигации (`$this->nav = array('edittext' => 'Edit', 'edit' => true);`), сценарий редактирования прямо в ячейке поддерживается?
3. Не нашел в документации упоминания о том, что для того, чтобы выброшенные на сервере исключения обрабатывались на клиенте, нужно подключить `jqgrid-ext.js`. Может, стоит добавить?

**Comment by artvolk** [comment]
Да, использую последнюю версию отсюда, последний коммит b81811e16431db6ad165653eb8529c775ba1258a (это правильный способ указать ревизию в git?)

**Comment by littleK0i** [comment]
К сожалению, Inline editing так просто не работает. Насколько я помню, там нельзя подставить функцию обработки ошибок по умолчанию. И нужно каждый раз явно прописывать errorHandler.

Возможно, сейчас уже добавили такую возможность.
Либо еще можно сделать свою обертку над editGridRow, которая везде будет добавлять общие handler'ы.

В целом, у inline редактирования еще много всяких мелких проблем, поэтому стараюсь его пореже использовать.

Упоминание о подключении js добавлю.

**Comment by artvolk** [comment]
Набросал такой вариант, который решил мою задачу, как тебе?

``` javascript
    onSelectRow: function(id) { 
        if (id && id !== $.jgrid.ext.lastSelected)
        { 
            $grid.jqGrid('restoreRow', $.jgrid.ext.lastSelected); 
            $.jgrid.ext.lastSelected = id; 
            $grid.jqGrid('editRow', id, { 
                keys : true, 
                successfunc: function(response) {                   
                    $.jgrid.ext.lastSelected = null;                    
                    jsonResponse = $.parseJSON(response.responseText);              
                    if (jsonResponse && jsonResponse.error)
                    {
                        $.jgrid.ext.errorHandler(jsonResponse);
                        return false;
                    }
                    else
                    {
                        return true; // to reflect changes in grid
                    }                   
                }
            }); 
        } 

```

Если модифицировать `$.jgrid.ext.errorHandler()` так, чтобы он возвращал наличие ошибки (т.е. `true`/`false`), можно убрать `if` внутри.

P.S. Здесь используется именно _inline editing_, у них ещё оказывает _cell editing_ есть :) 
Всего три типа: http://www.trirand.com/jqgridwiki/doku.php?id=wiki:common_rules#common_editing_properties
